clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -731,14 +727,10 @@ final void fail(long now, Throwable throwable) {
 
             // If the call has timed out, fail.
             if (calcTimeoutMsRemainingAsInt(now, deadlineMs) < 0) {
-                if (log.isDebugEnabled()) {
-                    log.debug("{} timed out at {} after {} attempt(s)", this, now, tries,
-                        new Exception(prettyPrintException(throwable)));
-                }
-                handleFailure(throwable);
+                failWithTimeout(now, throwable);
```

Comment:
Below, we have a case where we throw the underlying exception if we run out of retries. Is that intentional?

Reply:
Yeah, I was debating changing that one too. If retries are exhausted, do we consider that a timeout? I guess it's defensible.

Reply:
Right, if retries are exhausted and it's a retriable exception, then it seems like it should be a TimeoutException.